### PR TITLE
Feature/make classes abstract

### LINKED ISSUE #85: [v1.3.0] Make Gender.php and Countries.php abstract
Pretty self explanatory. No reason for these classes to be instantiatable.

### PR DESCRIPTION
**Pull request by bofalke**
Closes #85

I stumbled upon this awesome project during the Hacktoberfest. It would be great if you could label this PR according to their [recent update](https://hacktoberfest.digitalocean.com/hacktoberfest-update)

### CODE REVIEW

**Comment by AlexOlival** [comment]
Hi @bofalke, thank you for your contribution!
As stated by Digital Ocean in that link:
>Submitted in a public repo AND (
  The PR is labelled as hacktoberfest-accepted by a maintainer OR
  Submitted in a repo with the hacktoberfest topic AND (
    The PR is merged OR
    The PR has been approved
  ) 

This repository already has the "hacktoberfest" topic, so it counts :smiley: worry not

**Review by AlexOlival — CHANGES_REQUESTED** [review]
Good job, but I just need you to do two more things:
1 - Target the v1.2.0 branch (apologies for the confusion - I named that branch wrong but soon it will be renamed to v1.3.0)
2 - Go ahead and create a test for making sure that the factories still work. Not only this guarantees that the `Countries` class being abstract is ok (which it is, but coverage is coverage) but it also means that future PRs will need to add a test to ensure the factories can correctly instantiate their countries. 

Thank you!

**Comment by AlexOlival** [comment]
I have fixed the branch name because it was erroneously targetting a tag.
Please change the merge target to release-1.3.0

File: tests/CitizenInformationExtractorFactoryTest.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+
+declare(strict_types=1);
```
Comment (AlexOlival) [inline tests/CitizenInformationExtractorFactoryTest.php:4]: I think we can lose these lines. Strict types would make more sense to be declared in the main entry point, but we're being good boys so far :angel:

File: tests/IdValidatorFactoryTest.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+
+declare(strict_types=1);
```
Comment (AlexOlival) [inline tests/IdValidatorFactoryTest.php:4]: Ditto

**Review by AlexOlival — CHANGES_REQUESTED** [review]
Couple more tweaks and we're golden :rocket:

File: tests/CitizenInformationExtractorFactoryTest.php (hunk 1 above)
Comment (bofalke) [reply tests/CitizenInformationExtractorFactoryTest.php:4]: Haha alright, that came straight from my default settings for new classes.

**Comment by AlexOlival** [comment]
@bofalke no need for those classes to be declared final too, remove that please :+1: 
You gave me some ideas on how we should improve the contributing guidelines to have these sorts of things explained. Better than to then have to look out for IDE and editor default :grin:

**Comment by bofalke** [comment]
Updated the classes to not be final

**Comment by AlexOlival** [comment]
@all-contributors please add @bofalke for code

Thank you very much!

**Review by AlexOlival — APPROVED** [review]

**Pull request merged**